sycl/include/CL/sycl/device.hpp

```diff
@@ -16,7 +16,6 @@
 #include <CL/sycl/platform.hpp>
 #include <CL/sycl/stl.hpp>
 
-#include <memory>
```

Comment:
`memory` was actually used for `shared_ptr`.

Reply:
It was deleted when I tested map. But memory is included in stl.hpp, that included higher. Do I need to return it?

Reply:
Yes. You can't rely on `CL/sycl/stl.hpp` to include `memory` header for you.

Reply:
Returned it